docs/api-reference/next/server.md

```diff
@@ -70,13 +70,59 @@ import type { NextFetchEvent } from 'next/server'
 
 ## NextResponse
 
-The `NextResponse` object is an extension of the native [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) interface, with the following added methods and properties:
+The `NextResponse` class extends the native [`Response`](https://developer.mozilla.org/en-US/docs/Web/API/Response) interface, with the following:
+
+### Public methods
+
+Public methods are available on an instance of the `NextResponse` class. Depending on your use case, you can create an instance and assign to a variable, then access the following public methods:
 
 - `cookies` - An object with the cookies in the `Response`
 - `cookie` - Set a cookie in the `Response`
+
+```ts
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+
+export function middleware(req: NextRequest) {
+  // create an instance of the class to access the public methods. This uses `next()`,
+  // you could use `redirect()` or `rewrite()` as well
+  const res = NextResponse.next()
+  // access the `cookies`
+  console.log(res.cookies['cookie-name'])
+  // set the `cookie`
+  res.cookie('hello', 'world')
+
+  return res
+}
+```
+
+### Static methods
+
+The following static methods are available on the `NextResponse` class directly:
+
 - `redirect()` - Returns a `NextResponse` with a redirect set
```

Comment:
What do you mean here @javivelasco ? To say only that `redirect()` is used to redirect headers?

Reply:
I mean that the "redirect" itself comes from a location header and a status code. Since it includes also a status code maybe it is just fine but I thought it might be more clear to include "headers" as it is a `location` header :)